api/github_api/util/SceGithubApiHandler.js

```diff
@@ -0,0 +1,142 @@
+   * e.g. [ {user: "user1", commits: 7}, {user: "user2", commits: 4}, ... ]
+   */
+  getContributorsInPastMonthFromRepo(repoName) {
+    return fetch(
+      `https://api.github.com/repos/SCE-Development/${repoName}/commits` +
+      `?client_id=${this.clientId}&client_secret=${this.clientSecret}`
+    )
+      .then(response => response.json())
+      .then(data => {
+        if(!data.length) {
+          return {};
+        }
+        const endDate = new Date(Date.now());
+        const startDate = new Date(endDate - (2592000000)); // Subtract 30 days
+        const commitCount = {};
+        const avatarUrls = {};
+        const contributors = [];
+
+        data.forEach(commit => {
+          const commitDate = new Date(commit.commit.author.date);
+          const username = commit.author.login;
+
+          if(!avatarUrls[username]) {
+            avatarUrls[username] = commit.author.avatar_url;
+          }
+          if(startDate <= commitDate && commitDate <= endDate) {
+            commitCount[username] = commitCount[username] + 1 || 1;
+          }
+        });
+
+        // Exclued bots
+        for(const user in commitCount) {
+          if(user.includes('[bot]')) {
+            continue;
+          } else {
+            contributors.push({
+              user: user,
+              avatarUrl: avatarUrls[user],
+              commits: commitCount[user]
+            });
+          }
+        }
+
+        return contributors.sort((user1, user2) =>
+          user2.commits - user1.commits
+        ).slice(0, 5);
+      })
+      .catch(_ => {
+        return false;
+      });
+  }
+
+  /**
+   * Retrieves all open Pull Requests for a given repo
+   * @param {String} repoName Name of SCE Github repo
+   * @returns {Array<Object>|boolean} List of Pull Requests
+   */
+  getPullRequestsFromRepo(repoName) {
+    return fetch(
+      `https://api.github.com/repos/SCE-Development/${repoName}/` +
+      `pulls?client_id=${this.clientId}&client_secret=${this.clientSecret}`
+    )
+      .then(response => response.json())
+      .then(data => {
+        const pullRequests = data.map(pr => {
+          return {
+            number: pr.number,
+            title: pr.title,
+            pullRequestUrl: pr.html_url
+          };
+        });
+        return pullRequests;
+      })
+      .catch(_ => {
+        return false;
+      });
+  }
+
+  /**
+   * Gets all commits merged into repos master branch
+   * @param {String} repoName Name of SCE Github repo
+   * @returns {Array<Object>|boolean} List containing commits
+   */
+  getCommitsFromRepo(repoName) {
+    return fetch(
+      `https://api.github.com/repos/SCE-Development/${repoName}/commits` +
+      `?client_id=${this.clientId}&client_secret=${this.clientSecret}`
+    )
+      .then(response => response.json())
+      .then(data => {
+        if(!data.length) {
+          return {};
+        }
+
+        return data.map(commit => {
```

Comment:
Assign this to a constant before returning, like what you did in pullRequests

Reply:
done